urlc175f05cb8_mchllbrt_PermCode_tgz-pJ8-permlib_examples_UniversalSeparableJ8/src/permlib/examples/SaganBijection.java

```diff
@@ -0,0 +1,173 @@
+                if (!t132.containsKey(ts)) {
+                    t132.put(ts, 0);
+                }
+                t132.put(ts, t132.get(ts)+1);
+            }
+            for(Permutation s : new Permutations(c231, n)) {
+                BTree ts = BTree.descendingTree(s);
+                if (!t231.containsKey(ts)) {
+                    t231.put(ts, 0);
+                }
+                t231.put(ts, t231.get(ts)+1);
+            }
+            for(BTree t : t132.keySet()) {
+                if (!t231.containsKey(t) || !t231.get(t).equals(t132.get(t))) {
+                    System.out.println(t132.get(t));
+                    System.out.println(t231.get(t));
+                    System.out.println("Bad tree " + t);
+                    return;
+                }
+            }
+            
+        }
+        
+    }
+    
+    private static void checkMaxPositions() {
+        int k = 2;
+        int m = 3;
+        int maxN = 12;
+        int[] p = new int[k+m+1];
+        int[] q = new int[k+m+1];
+        for(int i = 0; i < k; i++) {
+            p[i] = i;
+            q[i] = m + i;
+        }
+        p[k] = k+m;
+        q[k] = k+m;
+        for(int i = k+1; i < k+m+1; i++) {
+            p[i] = k+m+k-i;
+            q[i] = m+k-i;
+        }
+        PermutationClass c132 = new PermutationClass(new Permutation(p));
+        PermutationClass c231 = new PermutationClass(new Permutation(q));
+        for(int n = k+m+2; n < maxN; n++) {
+            int[] a = new int[n];
+            int[] b = new int[n];
+            for(Permutation s : new Permutations(c132,n)) {
+                a[maxPosition(s)]++;
+            }
+            for(Permutation s : new Permutations(c231,n)) {
+                b[maxPosition(s)]++;
+            }
+            System.out.println(Arrays.toString(a));
+            System.out.println(Arrays.toString(b));
+            if (!Arrays.equals(a, b)) break;
+        }
+    }
+    
+    private static void checkDesDistribution() {
+        int k = 2;
+        int m = 3;
+        int maxN = 12;
+        int[] p = new int[k+m+1];
+        int[] q = new int[k+m+1];
+        for(int i = 0; i < k; i++) {
+            p[i] = i;
+            q[i] = m + i;
+        }
+        p[k] = k+m;
+        q[k] = k+m;
+        for(int i = k+1; i < k+m+1; i++) {
+            p[i] = k+m+k-i;
+            q[i] = m+k-i;
+        }
+        PermutationClass c132 = new PermutationClass(new Permutation(p));
+        PermutationClass c231 = new PermutationClass(new Permutation(q));
+        for(int n = k+m+2; n < maxN; n++) {
+            HashMap<ArrayList<Integer>, Integer> a = new HashMap<>();
+            HashMap<ArrayList<Integer>, Integer> b = new HashMap<>();
+            for(Permutation s : new Permutations(c132, n)) {
+                ArrayList<Integer> ds = descents(s);
+                if (!a.containsKey(ds)) {
+                    a.put(ds, 0);
+                }
+                a.put(ds, a.get(ds)+1);
+            }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 128:134, 135:141.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.